### DISCUSSION TITLE
PictoPy Discussions!

### DISCUSSION THREAD

**Discussion by Pranav0-0Aggarwal** [AOSSIE-Org/PictoPy#28] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by deveshruttala** [comment]
Hello everyone,

I'm thrilled to be a part of this community. My name is Devesh, and I'm passionate about exploring how technology, particularly AI, can enhance our daily lives. In this community, I'm eager to engage in discussions about Pictopy's features, share tips and tricks for maximizing its utility, and brainstorm ideas for future enhancements. I'm excited to connect with fellow users and collaborate on making our photo organization experience even better with Pictopy!

Looking forward to getting to know all of you and exchanging insights!

**Comment by VanshSaxena04** [comment]
HI....I'm thrilled for the opportunity to be part of the open-source community at AOSSIE. It's exciting to connect with such a passionate group of developers and contribute to meaningful projects.

While exploring the AI section, I encountered an error popup and was wondering what it might be related to. Could you please guide me on whether it's a known issue or if there's anything I can do to help troubleshoot it?

Looking forward to contributing and learning with the community!

Best regards,
Vansh Saxena

**Comment by Aditya30ag** [comment]
Hello everyone ,
Aditya this side

✅ Issue #433 Resolved – Tauri Environment Detection & Safe API Integration

I've implemented a fix for the runtime errors that occurred when running the frontend in browser development mode (npm run dev). The issue stemmed from direct usage of Tauri APIs that are only available in the desktop environment, which caused TypeError: Cannot read properties of undefined errors.

🔧 Solution Highlights:
Environment Detection: Introduced isTauriEnvironment() to detect if the app is running in a Tauri context.

Safe API Wrappers: Created wrappers for Tauri APIs like invoke, dialog, updater, etc., to prevent crashes in browser mode.

Component Updates: Updated useUpdater.ts, FolderPicker, and other components to use these safe wrappers.

Graceful Fallbacks: Browser mode now handles unsupported features with clear messaging and no runtime errors.

Improved Developer Experience: Frontend can now be developed and tested in the browser without issues, while desktop mode remains fully functional.

🧪 Tested in both modes:

npm run dev – ✅ No crashes, limited features with messaging

npm run tauri dev – ✅ Full native functionality

**Comment by Sanskriti10247** [comment]
Hello @rahulharpal1603 & @Pranav0-0Aggarwal , I hope you’re doing well. I’m fairly new to the project and still in the process of learning my way around the codebase. I genuinely admire the work being done here and would love to contribute in a consistent and meaningful way over the long term. I am also hoping to apply for GSoC 2026, and I wanted to gently ask—if you don’t mind—whether this project might be proposed again for GSoC this year. Thank you so much for your time and for all the effort you put into the project.

**Comment by pranav-n29** [comment]
Hello @Pranav0-0Aggarwal  i would like to work on some issue please asign to me
